app/assets/javascripts/discourse/app/lib/plugin-api.js

```diff
@@ -1027,29 +1026,6 @@ class PluginApi {
     addComposerUploadHandler(extensions, method);
   }
 
-  /**
-   * Registers a pre-processor for file uploads
-   * See https://github.com/blueimp/jQuery-File-Upload/wiki/Options#file-processing-options
-   *
-   * Useful for transforming to-be uploaded files client-side
-   *
-   * Example:
-   *
-   * api.addComposerUploadProcessor({action: 'myFileTransformation'}, {
-   *    myFileTransformation(data, options) {
-   *      let p = new Promise((resolve, reject) => {
-   *        let file = data.files[data.index];
-   *        console.log(`Transforming ${file.name}`);
-   *        // do work...
-   *        resolve(data);
-   *      });
-   *      return p;
-   * });
-   */
-  addComposerUploadProcessor(queueItem, actionItem) {
```

Comment:
If we remove this straight away, wouldn't it break plugins that are still relying on this API? I think we need to go through a proper deprecation cycle here.

Reply:
There are no plugins using this API, we actually only introduced it fairly recently in https://github.com/discourse/discourse/commit/2f60309b13c5808ee5d8389eaf7e2d9aeb191dcf and I've made sure to go through our all-the-* repos and replace the one or two things that did use it. Also this API is completely useless now that we are changing the composer to only use the uppy uploader. So I think we are fine to just remove it.

Reply:
I'm not sure if I consider 6 months fairly recently so I personally would err on the safe side and go through a deprecation cycle for this. However, if we're confident there are no external users of this API, we can take the risk and just remove without deprecation but it defeats the entire purpose of having a versioned API. 

> I've made sure to go through our all-the-* repos and replace the one or two things that did use it.

Did we replace it with another API call? We could in theory just make `addComposerUploadPrecessor` call the new API?

Reply:
Opened an internal discussion about this.